src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ..... } = action;
    
    switch (type) {
```